dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/client/ConfigCenterClient.java

```diff
@@ -375,15 +374,17 @@ public void refreshConfig(String configcenter, boolean wait) {
               }
               latch.countDown();
             });
-          } else {
+          }else if (rsp.statusCode() == HttpResponseStatus.NOT_MODIFIED.code()){
+            latch.countDown();
+          }else {
             rsp.bodyHandler(buf -> {
               LOGGER.error("Server error message is [{}].", buf);
               latch.countDown();
             });
             EventManager.post(new ConnFailEvent("fetch config fail"));
             LOGGER.error("Config refresh from {} failed.", configcenter);
           }
-        });
+        }).setTimeout((BOOTUP_WAIT_TIME -1 )* 1000);
```

Comment:
Thread control based on timeout may not be accurate. The pulling operation can be changed to blocking, but this code can not guarantee each pull run sequencially.

Reply:
I do not think so, I set the the time out of the sub thread to pull the setting from config-center is less than the main thread to wait . hence , the main thread will  wait until sub thread finish pull or throw a time out exception.  of course, it can gurantee each pull run sequentially